nixos/modules/services/computing/boinc/client.nix

```diff
@@ -6,6 +6,24 @@ let
   cfg = config.services.boinc;
   allowRemoteGuiRpcFlag = optionalString cfg.allowRemoteGuiRpc "--allow_remote_gui_rpc";
 
+  # A package providing the boinc_client that will actually be executed.
+  execPkg = if cfg.useFHSEnv then
+    pkgs.buildFHSUserEnv {
+      name = "boinc_client";
+      targetPkgs = p: [ cfg.package ] ++ optional cfg.virtualbox.enable cfg.virtualbox.package;
+      multiPkgs = p: optional cfg.gpu.nvidia.enable cfg.gpu.nvidia.package;
+      runScript = "/bin/boinc";
+      extraBuildCommands = ''
+        if [ -e lib/vendors ]; then
+          chmod +w etc
+          mkdir -p etc/OpenCL
+          ln -s $(realpath lib/vendors) etc/OpenCL/vendors
+          chmod -w etc
+        fi
+      '';
```

Comment:
I've already researched that a bit, it turns out OpenCL is broken on NixOS right now. Patch to fix it is,  luckily, trivial (and yes, includes using `/etc/OpenCL/vendors`; I'll make a PR soon.